File: docs/current/guides/snippets/use-services/use-hostnames-3/main.go
package main

import (
	"context"
	"fmt"
	"os"

	"dagger.io/dagger"
)

func main() {
	ctx := context.Background()

	// create Dagger client
	client, err := dagger.Connect(ctx, dagger.WithLogOutput(os.Stderr))

	if err != nil {
		panic(err)
	}
	defer client.Close()

	// get hostname of service container via API
	val, err := client.Container().
		From("python").
		WithExec([]string{"python", "-m", "http.server"}).
		Hostname(ctx)

	if err != nil {
		panic(err)
	}

	fmt.Println(val)
}

File: docs/current/guides/snippets/use-services/use-hostnames-3/main.py
import sys

import anyio

import dagger


async def main():
    # create Dagger client
    async with dagger.Connection(dagger.Config(log_output=sys.stderr)) as client:
        # get hostname of service container via API
        val = await (
            client.container()
            .from_("python")
            .service(["python", "-m", "http.server"])
            .hostname()
        )

    print(val)


anyio.run(main)
